src/pages/[...slug].astro

```diff
@@ -1,38 +1,117 @@
 ---
+
 import { getCollection } from "astro:content";
 
 import Layout from "@/layouts/base.astro";
-import { sys } from "typescript";
 
-export async function getStaticPaths() {
-    const documents = await getCollection("documents");
+export const discordCache = new Map<string,{username:string,pfp:string}|undefined>();
+export const githubCache = new Map<string,{username:string,pfp:string,url:string}|undefined>();
```

Comment:
Use `T | null` as `Map<T>.get(...)` returns `T | undefined` - `null` should be used as the special-case value. This is detailed in a separate comment further into the file.